gnomad_qc/v4/create_release/create_combined_faf_release_ht.py

```diff
@@ -0,0 +1,455 @@
+      independence depending on min cell count.
+    - Cochran–Mantel–Haenszel test -- stratified test of independence for 2x2xK
+      contingency tables.
+
+"""
+import argparse
+import logging
+from typing import Dict, List, Set, Tuple
+
+import hail as hl
+from gnomad.resources.grch38.gnomad import POPS, POPS_TO_REMOVE_FOR_POPMAX
+from gnomad.utils.annotations import faf_expr, merge_freq_arrays, pop_max_expr
+from gnomad.utils.release import make_faf_index_dict, make_freq_index_dict
+from gnomad.utils.slack import slack_notifications
+from statsmodels.stats.contingency_tables import StratifiedTable
+
+from gnomad_qc.resource_utils import (
+    PipelineResourceCollection,
+    PipelineStepResourceCollection,
+)
+from gnomad_qc.slack_creds import slack_token
+from gnomad_qc.v4.resources.annotations import (
+    get_combined_frequency,
+    get_freq_comparison,
+)
+from gnomad_qc.v4.resources.basics import get_logging_path
+from gnomad_qc.v4.resources.release import get_combined_faf_release, release_sites
+
+logging.basicConfig(format="%(levelname)s (%(name)s %(lineno)s): %(message)s")
+logger = logging.getLogger("compute_combined_faf")
+logger.setLevel(logging.INFO)
+
+
+def extract_freq_info(
+    ht: hl.Table,
+    pops: List[str],
+    faf_pops: List[str],
+    prefix: str,
+) -> hl.Table:
+    """
+    Extract frequencies and FAF for populations in `pops` and `faf_pops` respectively.
+
+    .. note::
+
+        The reduced frequency arrays include adj and raw in addition to the `pops`.
+
+    The following annotations are filtered and renamed:
+        - freq: {prefix}_freq
+        - faf: {prefix}_faf
+        - grpmax: {prefix}_grpmax
+
+    The following global annotations are filtered and renamed:
+        - freq_meta: {prefix}_freq_meta
+        - faf_meta: {prefix}_faf_meta
+
+    This only keeps variants that PASS filtering.
+
+    :param ht: Table with frequency and FAF information.
+    :param pops: List of populations to include in the reduced frequency arrays.
+    :param faf_pops: List of populations to include in the reduced FAF arrays.
+    :param prefix: Prefix to add to each of the filtered annotations.
+    :return: Table with filtered frequency and FAF information.
+    """
+    logger.info("Filtering Table to only variants that are PASS...")
+    ht = ht.filter(hl.len(ht.filters) == 0)
+
+    def _get_pop_meta_indices(
+        meta: hl.ArrayExpression, pop_list: List[str]
+    ) -> Tuple[List[int], List[dict]]:
+        """
+        Keep full gnomAD callset adj [0], raw [1], and ancestry-specific adj frequencies.
+
+        :param meta: Array of frequency metadata.
+        :param pop_list: List of populations to keep.
+        :return: Indices of populations to keep and their metadata.
+        """
+        meta = hl.eval(meta)
+        idx = [0, 1]
+        idx.extend(
+            [i for i, m in enumerate(meta) if m.get("pop") in pop_list and len(m) == 2]
+        )
```

Comment:
at least for genomes, there is no 'raw' for faf so '{'group': 'adj', 'pop': 'nfe'}' is repeated twice for index 1?

Reply:
I fixed this one.